include/rocksdb/options.h

```diff
@@ -269,6 +269,17 @@ struct ColumnFamilyOptions : public AdvancedColumnFamilyOptions {
   // Dynamically changeable through SetOptions() API
   uint64_t max_bytes_for_level_base = 256 * 1048576;
 
+  // If non-zero, compactions will periodically refresh the snapshot list. The
+  // delay for the first refresh is snap_refresh_nanos nano seconds and
+  // exponentially increases afterwards. When having many short-lived snapshots,
+  // this option helps reducing the cpu usage of long-running compactions. The
+  // feature is disabled when max_subcompactions is greater than one.
+  //
+  // Default: 0.5s
+  //
+  // Dynamically changeable through SetOptions() API
+  uint64_t snap_refresh_nanos = 500 * 1000 * 1000;  // 0.5s
```

Comment:
This feels long to me. I would believe it's common for requests to finish in milliseconds. So after several ms, majority of snapshots should be gone. On the other hand, many compactions finish in sub-seconds, so 0.5s can be a big portion of comapction. Do we consider to start with something like 5 ms?

Reply:
The feature has pros and cons and I am afraid if the snap_refresh_nanos is too short the cons might back fire. The cons is the cost of copying the entire list of snapshots, which could be 100k of them, to the vector. Not only it consumes cpu, it locks db_mutex. Moreover for the feature to be beneficial most of the snapshots should be already released by the time that refresh is called. I figured for the starter, 0.5s should be small enough to show benefit in 20s+ compactions and yet far from average snapshot lifetime to stay away from the costs.